subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/DefaultIvyArtifactName.java

```diff
@@ -33,12 +33,18 @@
     private final int hashCode;
 
     public static DefaultIvyArtifactName forPublishArtifact(PublishArtifact publishArtifact) {
-        String name = publishArtifact.getName();
-        if (name == null) {
-            name = publishArtifact.getFile().getName();
+        // TODO: 18791 There must be a better way to handle the failure of JacocoPluginIntegrationTest#"reports miss configuration of destination file"
+        // after adding the destination file as an artifact in JacocoPlugin - that test uses a provider { null } for the destination file.
+        try {
+            String name = publishArtifact.getName();
+            if (name == null) {
+                name = publishArtifact.getFile().getName();
+            }
+            String classifier = GUtil.elvis(publishArtifact.getClassifier(), null);
```

Comment:
This is so odd - I wonder why the original author didn't use `GUtil#elvis` to assign `name`?

So it's not your fault but I suggest removing the usage of `GUtil`, since we also don't add references to `g.l.Closure` when we can help it.